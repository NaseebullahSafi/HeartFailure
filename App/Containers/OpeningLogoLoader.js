import React, { Component } from 'react'
import {View, StyleSheet, TouchableHighlight, Text} from "react-native";
import Ionicons from 'react-native-vector-icons/Feather';
import Fade from '../Components/Fade'

export default class OpeningLogoLoader extends Component {

  constructor() {
    super();
    this.state = {
      loading: false
    }
  }

  componentDidMount() {
    this.setState({loading: true});
  }

  render() {
    return(
      <View style={styles.container}>
        <Fade visible={this.state.loading}>
          <Ionicons name="activity" size={33} color="#8F9CAE" />
        </Fade>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center', // Used to set Text Component Vertically Center
    alignItems: 'center' // Used to set Text Component Horizontally Center
  }
});
