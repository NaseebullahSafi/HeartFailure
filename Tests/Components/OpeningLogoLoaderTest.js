// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// the component that will be tested
import OpeningLogoLoader from "../../App/Containers/OpeningLogoLoader";

Enzyme.configure({ adapter: new Adapter() });

describe('Testing <OpeningLogoLoader /> component', () => {
  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <OpeningLogoLoader />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('Should mount the component', () => {
    const wrapper = shallow(<OpeningLogoLoader />);
    const inst = wrapper.instance().constructor();
    expect(inst)
  });
});

