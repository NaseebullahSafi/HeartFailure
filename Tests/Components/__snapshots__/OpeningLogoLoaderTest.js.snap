// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <OpeningLogoLoader /> component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 1.1,
          },
        ],
      }
    }
    visible={true}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#8F9CAE",
            "fontSize": 33,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¤€
    </Text>
  </View>
</View>
`;
